e2e/staticpvc.go

```diff
@@ -159,3 +159,116 @@ func validateRBDStaticPV(f *framework.Framework, appPath string, isBlock bool) e
+	fsID, e := execCommandInPod(f, "ceph fsid", rookNamespace, &listOpt)
+	if e != "" {
+		return fmt.Errorf("failed to get fsid from ceph cluster %s", e)
+	}
+	// remove new line present in fsID
+	fsID = strings.Trim(fsID, "\n")
+
+	// 4G in bytes
+	size := "4294967296"
+
+	// create subvolumegroup, command will work even if group is already present.
+	cmd := fmt.Sprintf("ceph fs subvolumegroup create %s %s", fsName, groupName)
+
+	_, e = execCommandInPod(f, cmd, rookNamespace, &listOpt)
+	if e != "" {
+		return fmt.Errorf("failed to create subvolumegroup %s", e)
+	}
+
+	// create subvolume
+	cmd = fmt.Sprintf("ceph fs subvolume create %s %s %s --size %s", fsName, cephFsVolName, groupName, size)
+	_, e = execCommandInPod(f, cmd, rookNamespace, &listOpt)
+	if e != "" {
+		return fmt.Errorf("failed to create subvolume %s", e)
+	}
+
+	opt["clusterID"] = fsID
+	opt["fsName"] = fsName
+	opt["staticVolume"] = "true"
+	opt["rootPath"] = "/volumes/" + groupName + "/" + cephFsVolName
+
+	// create secret
+	const userID = "admin"
+	secret := getSecret(scPath)
+	adminKey, e := execCommandInPod(f, "ceph auth get-key client.admin", rookNamespace, &listOpt)
+	if e != "" {
+		return fmt.Errorf("failed to get adminKey %s", e)
+	}
+	secret.StringData["userID"] = userID
+	secret.StringData["userKey"] = adminKey
+	secret.Name = secretName
+	_, err := c.CoreV1().Secrets("default").Create(context.TODO(), &secret, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create secret, error %v", err)
+	}
+
+	pv := getStaticPV(pvName, pvName, "4Gi", secretName, "default", sc, "cephfs.csi.ceph.com", false, opt)
+	_, err = c.CoreV1().PersistentVolumes().Create(context.TODO(), pv, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create PV, error %v", err)
+	}
+
+	pvc := getStaticPVC(pvcName, pvName, size, "default", sc, false)
+
+	_, err = c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Create(context.TODO(), pvc, metav1.CreateOptions{})
+	if err != nil {
+		return fmt.Errorf("failed to create PVC, error %v", err)
+	}
+	// bind pvc to app
+	app, err := loadApp(appPath)
+	if err != nil {
+		return fmt.Errorf("failed to load app, error %v", err)
+	}
+
+	app.Namespace = "default"
+	app.Spec.Volumes[0].PersistentVolumeClaim.ClaimName = pvcName
+	err = createApp(f.ClientSet, app, deployTimeout)
+	if err != nil {
+		return fmt.Errorf("failed to create pod, error %v", err)
+	}
+
+	err = deletePod(app.Name, "default", f.ClientSet, deployTimeout)
+	if err != nil {
+		return fmt.Errorf("failed to delete pod, error %v", err)
+	}
+
+	err = c.CoreV1().PersistentVolumeClaims(pvc.Namespace).Delete(context.TODO(), pvc.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	err = c.CoreV1().PersistentVolumes().Delete(context.TODO(), pv.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
+
+	err = c.CoreV1().Secrets("default").Delete(context.TODO(), secret.Name, metav1.DeleteOptions{})
+	if err != nil {
+		return err
+	}
```

Comment:
We need to clean up the subvolumegroup and subvolume created for this testing. better to create different subvolume group and see it works

Reply:
fixed